src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -617,7 +617,7 @@ protected override long ExecuteCmd(DocumentsOperationContext context)
                         {
                             count++;
                             var changeVector = $"{ChangeVectorParser.RaftTag}:{count}-{Database.DatabaseGroupId}";
-                            if (options.DisableAtomicDocumentWrites == false)
+                            if (options.DisableAtomicDocumentWrites == false && Database.ClusterTransactionId != null)
```

Comment:
I think the issue here is more complex.
If we reach here I assume we will create the atomic guard, but will mess the change vector by not include the `TRXN` element, which will cause mis-synchronization.. and we will be unable to modify this document. 

I think we have few issues to address here:
1. If we have the atomic guard and we reached this code with `ClusterTransactionId` as `null` - not sure yet what to do
2. In the state machine verify we have  `ClusterTransactionId` when `DisableAtomicDocumentWrites` is `false` (if not, we might able to create it, but has to be consistent value across all nodes)
3. This also relevant for 5.2, no?

Reply:
If I remember right the `ClusterTransactionId` should be created and updated in the following code.
We did it as part of a raft command (the `ClusterTransactionCommand`) so it will be consistent in all nodes.
https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs#L139
https://github.com/ravendb/ravendb/blob/v5.3/src/Raven.Server/ServerWide/ClusterStateMachine.cs#L978

Reply:
What happened if we created the cluster-wide command in the old server version (so we don't have this `ClusterTransactionId`), but execute it on the new version? 
I would expect `options.DisableAtomicDocumentWrites` to be `null` in that case and not `false`